src/deepsparse/pipeline.py

```diff
@@ -146,7 +162,7 @@ def __init__(
                 )
 
         self._engine_args = dict(
-            batch_size=batch_size,
+            batch_size=self._batch_size or 1,  # bs=1 for dynamic batch
```

Comment:
Does the comment mean that when using thread pool executor batch size has to be 1? What happens if user specifies batch size > 1 and a thread pool executor?

Reply:
dynamic batch wouldn't be supported, but async runs would be

Reply:
this brings up a good point though, we could probably add some generic support for batch splitting as long as the given batch size is divisible by the engine batch size

Reply:
Ah gotcha, I think the interaction between the two modes should be captured, it sounds like:

1. batch_size=None
    a. executor is specified => async runs
    b. executor is not specified => sync static batch
2. batch_size=1
    a. executor is specified => sync dynamic batch, _executor is not used_
    b. executor is not specified => sync dynamic batch
3. batch_size > 1 (does it have to be engine_batch_size?)
    a. executor is specified => sync dynamic batch, _executor is not used_
    b. executor is not specified => sync static batch

Is that correct?

Reply:
close, currently dynamic batch is only executed when bs=None